test/internal/request/complement.spec.js

```diff
@@ -0,0 +1,113 @@
+var expect = require("chai").expect;
+var complement = require("../../../lib/request/complement");
+var findPartialIntersections = require("../../../lib/request/complement").findPartialIntersections;
+
+describe("complement", function() {
+    it("returns null if no paths are provided", function() {
+        expect(complement([], [], {})).to.be.null;
+    });
+
+    it("returns null if no deduping was possible", function() {
+        var paths = [["videos", 0, "title"]];
+        expect(complement(paths, paths, {})).to.be.null;
+    });
+
+    it("returns the complement and intersection consisting of paths that can be partially deduped", function() {
+        var partialMatchingPath = ["videos", [0, 1], "title"];
+        var paths = [partialMatchingPath];
+        var pathTree = { "3": { videos: { "0": { title: null } } } };
+
+        expect(complement(paths, paths, pathTree)).to.deep.equal([
+            [["videos", 0, "title"]],
+            [["videos", 1, "title"]],
+            [["videos", 1, "title"]]
+        ]);
+    });
+});
+
+describe("findPartialIntersections", function() {
+    var matchingPath;
+    var matchingPathTree;
+
+    beforeEach(function() {
+        matchingPath = ["lolomo", 123, "summary"];
+        matchingPathTree = { lolomo: { "123": { summary: null } } };
```

Comment:
The quotes are unnecessary around `"123"`. All object keys are strings.

```suggestion
        matchingPathTree = { lolomo: { 123: { summary: null } } };
```

Eslint has a rule to catch this. It would be great if it were enabled here.

Reply:
Eslint is pretty out of date too. I'll see if I can update it.